pkg/nodeinfo/interface.go

```diff
@@ -0,0 +1,54 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package nodeinfo
```

Comment:
This package is a good start. But it won't just be node info. I think we'll use it to get depot details and node subscriptions. Since these all come from the vertica catalog, maybe call it catalog?

Reply:
Got you! Will we be getting them from endpoints too?